client/src/components/ImageGallery/ImageGallery.js

```diff
@@ -0,0 +1,112 @@
+import React, { useState, useCallback } from 'react';
+import { Swiper, SwiperSlide } from 'swiper/react';
+import ImageViewer from 'react-simple-image-viewer';
+
+// eslint-disable-next-line import/no-unresolved
+import 'swiper/css';
+
+// eslint-disable-next-line import/no-unresolved
+import './Swiper.css';
+
+import { Autoplay, FreeMode, Mousewheel, Scrollbar, Zoom, Thumbs, Navigation } from 'swiper';
+
+export default function ImageGallery() {
+  const [currentImage, setCurrentImage] = useState(0);
+  const [isViewerOpen, setIsViewerOpen] = useState(false);
+  const [thumbsSwiper, setThumbsSwiper] = useState(null);
+
+  const image = [
+    'http://placeimg.com/1200/800/nature',
+    'http://placeimg.com/800/1200/nature',
+    'http://placeimg.com/800/1200/nature',
+    'http://placeimg.com/1920/1080/nature',
+    'http://placeimg.com/1500/500/nature',
+    'http://placeimg.com/1500/500/nature',
+    'http://placeimg.com/800/1200/nature',
+    'http://placeimg.com/1500/500/nature',
+    'http://placeimg.com/1200/800/nature',
+    'http://placeimg.com/1200/800/nature',
+    'http://placeimg.com/1200/800/nature',
+  ];
+
+  const openImageViewer = useCallback((index) => {
+    setCurrentImage(index);
+    setIsViewerOpen(true);
+  }, []);
+
+  const closeImageViewer = () => {
+    setCurrentImage(0);
+    setIsViewerOpen(false);
+  };
+
+  return (
+    <div className="c">
+      <div id="p" className="swiper-button-prev" />
+      <div id="n" className="swiper-button-next" />
+      <Swiper
+        spaceBetween={5}
+        zoom
+        loop
+        longSwipes
+        autoplay={{
+          delay: 2500,
+          disableOnInteraction: true,
+          pauseOnMouseEnter: true,
+        }}
+        thumbs={{ swiper: thumbsSwiper }}
+        watchSlidesProgress
+        navigation={{
+          nextEl: '#n',
+          prevEl: '#p',
+        }}
+        mousewheel
+        modules={[Autoplay, Mousewheel, FreeMode, Scrollbar, Zoom, Navigation, Thumbs]}
+        slidesPerView="auto"
+        className="mySwiper2"
+      >
+        {image.map((src, index) => (
+          // eslint-disable-next-line react/no-array-index-key
+          <SwiperSlide key={index}>
```

Comment:
its a very bad practice to use the index as a key, this might easily break down the whole logic in the future. please use mathRandom or react id to generate unique keys.

Reply:
fixed